apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java

```diff
@@ -100,12 +103,37 @@ public static void delete(File file) throws IOException {
    }
 
    /**
+    * @return Localized name for the "Everyone" Windows principal.
+    */
+   public static String getWindowsEveryonePrincipalName() {
+      Runtime rt = Runtime.getRuntime();
+      Process pr = null;
+      try {
+         pr = rt.exec("whoami /groups | find \"S-1-1-0\"");
+      } catch (IOException e) {
+         return "Everyone";
+      }
+      try {
+         pr.waitFor();
+      } catch (InterruptedException e) {
+         e.printStackTrace();
+      }
+      try ( InputStreamReader reader = new InputStreamReader(pr.getInputStream(), Charsets.UTF_8)) {
+         return CharStreams.toString(reader).split(" ")[0];
+      } catch (IOException e) {
+         return "Everyone";
+      }
```

Comment:
Could you explain this a bit? Why do we need the three blocks here? Could this be simplified into one block?